### PR TITLE
[NEW] Enable twilio whatsapp template

### LINKED ISSUE #1385: WhatsApp interactive messages
Twilio has a new 'content builder that will create 'templates' that will render as 'interactive messages' in whatsApp.

Does the twilio node api package that RC uses support sending messages with templates?

Design:

- New action payload from CX, that would describe whatsApp template info that would be sent properly from RC, using the twilio API, to send the template/interactive message to the whatsApp visitor. 

@bhardwajaditya can you please look into this and let me know if it's feasible, and how much work this will involve.

### PR DESCRIPTION
**Pull request by bhardwajaditya**
## Proposed changes (including videos or screenshots)
Closes: https://github.com/WideChat/Rocket.Chat/issues/1385

### CODE REVIEW

File: apps/meteor/app/sms/server/services/twilio.js [hunk 1]
```diff
@@ -167,6 +179,9 @@ class Twilio {
 			body,
 			...(mediaUrl && { mediaUrl }),
 			...(persistentAction && { persistentAction }),
+			...(contentSid && { contentSid }),
```
Comment (ear-dev) [inline apps/meteor/app/sms/server/services/twilio.js:186]: It looks like you are adding the template attributes in all cases, but sometimes they will be empty?  Am I understanding that correctly, or are those three dots a conditional or something?  If so, are we sure they are NOOP and won't affect non-template messages?

Comment (ear-dev) [reply apps/meteor/app/sms/server/services/twilio.js:186]: And... in the case where we are using the template, is the body empty?  And that also has no effect?

Comment (bhardwajaditya) [reply apps/meteor/app/sms/server/services/twilio.js:186]: Yes if that value does not exist it won't add these so it's conditional, having empty body doesn't have any effect that's fine

**Comment by ear-dev** [comment]
@bhardwajaditya please add logging that shows exactly what is in the call we make to twilio.  

- Include all the parameters...... twilio team has suggested that they are only getting the **to** and **from** fields.
- Since we use the twilio library method client.messages.create() it could be possible that the library is not accepting our extra params.  Is there a logging function upon POST that we can use to see exactly what it is sending.

**Comment by ear-dev** [comment]
@bhardwajaditya Looking again at the twilio docs it seems that they are showing a way to log what the twilio library is doing by using the .then() part?

client.messages
      .create({
         contentSid: 'HX...XXX',
         from: 'MG...XXX',
         contentVariables: JSON.stringify({
           1: 'Name'
         }),
         to: 'whatsapp:+18551234567'
       })
      .then(message => console.log(message.sid));

FYI - in regard to the twilio nodejs library, the rep sent me this link for modifications:  https://github.com/twilio/twilio-node/blob/main/advanced-examples/custom-http-client.md

**Pull request merged**